core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -186,6 +186,10 @@ private[deploy] class Master(
         val fsFactory =
           new FileSystemRecoveryModeFactory(conf, serializer)
         (fsFactory.createPersistenceEngine(), fsFactory.createLeaderElectionAgent(this))
+      case "ROCKSDB" =>
```

Comment:
Is this configuration item required to be `ROCKSDB`? Is it not allowed to configure it as `rocksdb` or `Rocksdb` now? （Perhaps this is a legacy issue?）

Reply:
Yes, it's a legacy behavior which `spark.deploy.recoveryMode` always assumes all upper cases from Spark 0.8.1.
- NONE, ZOOKEEPER, FILESYSTEM, CUSTOM